ui/lib/kv/addon/components/page/list.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { getOwner } from '@ember/application';
+import { ancestorKeysForKey } from 'core/utils/key-utils';
+import errorMessage from 'vault/utils/error-message';
+import { pathIsDirectory } from 'kv/utils/kv-breadcrumbs';
+
+/**
+ * @module List
+ * ListPage component is a component to show a list of kv/metadata secrets.
+ *
+ * @param {array} secrets - An array of models generated form kv/metadata query.
+ * @param {string} backend - The name of the kv secret engine.
+ * @param {string} pathToSecret - The directory name that the secret belongs to ex: beep/boop/
+ * @param {string} pageFilter - The input on the kv-list-filter. Does not include a directory name.
+ * @param {string} filterValue - The concatenation of the pathToSecret and pageFilter ex: beep/boop/my-
+ * @param {boolean} noMetadataListPermissions - true if the return to query metadata LIST is 403, indicating the user does not have permissions to that endpoint.
+ * @param {array} breadcrumbs - Breadcrumbs as an array of objects that contain label, route, and modelId. They are updated via the util kv-breadcrumbs to handle dynamic *pathToSecret on the list-directory route.
+ * @param {string} routeName - Either list or list-directory.
+ */
+
+export default class KvListPageComponent extends Component {
+  @service flashMessages;
+  @service router;
+  @service store;
+
+  @tracked secretPath;
+
+  get mountPoint() {
+    // mountPoint tells transition where to start. In this case, mountPoint will always be vault.cluster.secrets.backend.kv.
+    return getOwner(this).mountPoint;
+  }
+
+  get buttonText() {
+    return pathIsDirectory(this.secretPath) ? 'View directory' : 'View secret';
```

Comment:
```suggestion
    return pathIsDirectory(this.secretPath) ? 'View list' : 'View secret';
```

Reply:
I'll make this change in the clean up PR I'm working on. Keep it all in the same commit.